src/it/scala/com/contxt/kinesis/KinesisResourceManager.scala

```diff
@@ -45,72 +66,112 @@ object KinesisResourceManager {
+  def withKinesisClient[A](regionName: String)(closure: KinesisClient => A): A = {
     val client = mkKinesisClient(regionName)
-    try {
-      closure(client)
-    }
-    finally {
-      client.shutdown()
-    }
+    closure(client)
   }
 
-  def withDynamoDbClient[A](closure: DynamoDB => A): A = {
-    val client = AmazonDynamoDBClientBuilder.standard().build()
-    val dynamoDb = new DynamoDB(client)
-    try {
-      closure(dynamoDb)
-    }
-    finally {
-      dynamoDb.shutdown()
-    }
+  def withDynamoDbClient[A](closure: DynamoDbClient => A): A = {
+    val dynamoDb = DynamoDbClient.builder().build()
+    closure(dynamoDb)
   }
 
-  def updateDynamoDbTableWithRate(tableName: String, requetsPerSecond: Long): Unit = {
+  def updateDynamoDbTableWithRate(tableName: String, requestPerSecond: Long): Unit = {
     withDynamoDbClient { dynamoDb =>
-      val table = dynamoDb.getTable(tableName)
-      table.waitForActive()
-      table.updateTable(new ProvisionedThroughput(requetsPerSecond, requetsPerSecond))
-      table.waitForActive()
+      waitForTableToBecomeAvailable(tableName, dynamoDb)
+      dynamoDb.updateTable(
+        UpdateTableRequest
+          .builder()
+          .tableName(tableName)
+          .provisionedThroughput(
+            ProvisionedThroughput
+              .builder()
+              .readCapacityUnits(requestPerSecond)
+              .writeCapacityUnits(requestPerSecond)
+              .build()
+          )
+          .build()
+      )
+      waitForTableToBecomeAvailable(tableName, dynamoDb)
     }
   }
 
-  private def mkKinesisClient(regionName: String): AmazonKinesis = {
-    val clientBuilder = AmazonKinesisClientBuilder.standard()
-    clientBuilder.setRegion(regionName)
-    clientBuilder.setCredentials(CredentialsProvider)
-    clientBuilder.setClientConfiguration(PredefinedClientConfigurations.defaultConfig())
-    clientBuilder.build()
+  private def mkKinesisClient(regionName: String): KinesisClient = {
+    KinesisClient
+      .builder()
+      .region(Region.of(regionName))
+      .credentialsProvider(CredentialsProvider)
+      .build()
   }
 
   private def deleteDynamoDbTable(tableName: String): Unit = {
     withDynamoDbClient { dynamoDb =>
       try {
-        val table = dynamoDb.getTable(tableName)
-        table.waitForActive()
-        table.delete()
-        table.waitForDelete()
-      }
-      catch {
+        waitForTableToBecomeAvailable(tableName, dynamoDb)
+        dynamoDb.deleteTable(
+          DeleteTableRequest
+            .builder()
+            .tableName(tableName)
+            .build()
+        )
+      } catch {
         case NonFatal(e) => // Ignore.
       }
     }
   }
+
+  private def waitForTableToBecomeAvailable(tableName: String, dynamoDb: DynamoDbClient): Unit = {
+    println("Waiting for " + tableName + " to become ACTIVE...")
+    val startTime = System.currentTimeMillis
+    val endTime = startTime + (10 * 60 * 1000)
```

Comment:
Is DynamoDb really that slow? 10 minute wait time seems excessive.

Reply:
10minute is the maximum timeout, this loop can terminate early.